src/plugins/intel_gna/gna_plugin.cpp

```diff
@@ -270,90 +272,62 @@ void GNAPlugin::ExportScores(void *ptr_dst,
-                  float scaleFactor,
-                  intel_dnn_orientation_t orientation,
-                  uint32_t num_frames,
-                  uint32_t num_group,
-                  uint32_t num_vector_elements,
-                  uint32_t num_vector_stride) {
-    if (orientation == kDnnInterleavedOrientation) {
-        // TODO : fix that as well
-        if (input_precision == Precision::U8) {
-            auto src = reinterpret_cast<const uint8_t *>(ptr_src);
-            if (!gnaFlags->input_low_precision) {
-                auto dst = reinterpret_cast<int16_t*>(ptr_dst);
-                copyInputData(dst, src, num_frames, num_group, num_vector_elements, num_vector_stride, orientation, scaleFactor);
-            } else {
-                auto dst = reinterpret_cast<int8_t*>(ptr_dst);
-                copyInputData(dst, src, num_frames, num_group, num_vector_elements, num_vector_stride, orientation, scaleFactor);
-            }
-        } else if (input_precision.size() == 2) {
-            auto src = reinterpret_cast<const int16_t *>(ptr_src);
-            if (!gnaFlags->input_low_precision) {
-                auto dst = reinterpret_cast<int16_t*>(ptr_dst);
-                copyInputData(dst, src, num_frames, num_group, num_vector_elements, num_vector_stride, orientation, scaleFactor);
-            } else {
-                auto dst = reinterpret_cast<int8_t*>(ptr_dst);
-                copyInputData(dst, src, num_frames, num_group, num_vector_elements, num_vector_stride, orientation, scaleFactor);
-            }
-        } else if (input_precision.size() == 4) {
-            if (!gnadevice) {
-                auto dst = reinterpret_cast<float *>(ptr_dst);
-                auto src = reinterpret_cast<const float *>(ptr_src);
-                copyInputData(dst, src, num_frames, num_group, num_vector_elements, num_vector_stride, orientation, scaleFactor);
-            } else {
-                auto src = reinterpret_cast<const float *>(ptr_src);
-                if (!gnaFlags->input_low_precision) {
-                    auto dst = reinterpret_cast<int16_t*>(ptr_dst);
-                    copyInputData(dst, src, num_frames, num_group, num_vector_elements, num_vector_stride, orientation, scaleFactor);
-                } else {
-                    auto dst = reinterpret_cast<int8_t*>(ptr_dst);
-                    copyInputData(dst, src, num_frames, num_group, num_vector_elements, num_vector_stride, orientation, scaleFactor);
-                }
-            }
+void GNAPlugin::ImportFrames(void *ptr_dst,
+                            const void *ptr_src,
+                            Precision input_precision,
+                            float scaleFactor,
+                            intel_dnn_orientation_t orientation,
+                            uint32_t num_frames,
+                            uint32_t num_group,
+                            uint32_t num_vector_elements,
+                            uint32_t num_vector_stride) {
```

Comment:
The cases are very similar.
Consider to make this function a template to eliminate the switch - case.
e.g., remove the input_precision parameter and make `ptr_src` cast type a template parameter.

Reply:
yes, but let do that refactoring in separate PR

Reply:
Sure